tools/evaluate_queries.cpp

```diff
@@ -207,33 +212,48 @@ int main(int argc, const char **argv)
-        } else {                                                                                 \
-            evaluate_queries<BOOST_PP_CAT(T, _index), wand_raw_index>(app.index_basename(),      \
-                                                                      app.wand_data_path(),      \
-                                                                      app.queries(),             \
-                                                                      app.thresholds_file(),     \
-                                                                      app.index_encoding(),      \
-                                                                      app.algorithm(),           \
-                                                                      app.k(),                   \
-                                                                      documents_file,            \
-                                                                      app.scorer(),              \
-                                                                      run_id);                   \
-        }                                                                                        \
+#define LOOP_BODY(R, DATA, T)                                                                \
+    }                                                                                        \
+    else if (app.index_encoding() == BOOST_PP_STRINGIZE(T))                                  \
+    {                                                                                        \
+        if (app.is_wand_compressed()) {                                                      \
+            if (quantized) {                                                                 \
+                evaluate_queries<BOOST_PP_CAT(T, _index), wand_uniform_index_quantized>(     \
+                    app.index_filename(),                                                    \
+                    app.wand_data_path(),                                                    \
+                    app.queries(),                                                           \
+                    app.thresholds_file(),                                                   \
+                    app.index_encoding(),                                                    \
+                    app.algorithm(),                                                         \
+                    app.k(),                                                                 \
+                    documents_file,                                                          \
+                    app.scorer(),                                                            \
+                    run_id);                                                                 \
+            } else {                                                                         \
```

Comment:
You can use the same trick with tuple as in `queries` to make it simpler and less error-prone.